### ISSUE TITLE
Allow curly braces "{" and "}" for folder names under /scripts/rest folder in Studio

### ISSUE THREAD

**Issue by alhambrav** [craftercms/craftercms#7047]
### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Duplicates

- [X] I have searched the existing issues

### Is your enhancement request related to a problem? Please describe.

We have a feature in Engine that allow you to have a folder in {NAME} format under `scripts/rest/` and these folders become templates for parameterized URLs.  When creating folders in Studio, folder names are restricted to letters, numbers, dash (-) and underscore (_).
It would be nice if we're allowed to create folder names with `{` and `}` (curly braces) under `scripts/rest/` in Studio instead of creating the folders outside of Studio.

<img width="436" alt="Screenshot 2024-07-18 at 4 35 57 PM" src="https://github.com/user-attachments/assets/8883a0fa-1579-48a3-a4da-e816049c3aa0">

### Describe the solution you'd like

Allow users to create folder names with `{` and `}` (curly braces) under `scripts/rest/` in Studio.

<img width="247" alt="Screenshot 2024-07-18 at 4 30 52 PM" src="https://github.com/user-attachments/assets/e15cff71-35dd-4870-a23d-c4ce7150e1e0">

**Comment by jmendeza** [comment]
Studio API does not block the creation of folder with curly braces. However, they will need to be url encoded if the name is sent as a QS param. e.g.:

```
curl --location --request POST 'http://localhost:8080/studio/api/1/services/api/1/content/create-folder.json?name=%7Bversion%7D&path=%2Fscripts%2Frest&site=ed1' \
--header 'Authorization: Bearer <TOKEN>'
```

We need to update the UI to allow those characters and encode them if present.

@sumerjabri @rart

**Comment by rart** [comment]
@sumerjabri the validation engine/logic changed drastically to accommodate this requirement. Please assign for thorough (re)testing.

**Comment by yacdaniel** [comment]
Verified fixed in `4.1.6-SNAPSHOT`
